### PR TITLE
BUG: Fix paren error in dual annealing accept_reject calculation

### LINKED ISSUE #10892: Possible typo in the a equation of optimize/dual_annealing
In the source code dual_annealing function, Line 260 writes
`pqv_temp = (self.acceptance_param - 1.0) * (
            e - self.energy_state.current_energy) / (
                self.temperature_step + 1.)`
I believe this is calculating the result of the brackets (from Ref. 4, Xiang Y, Gong XG. Efficiency of Generalized Simulated Annealing. Physical Review E, 62, 4473 (2000).):

where you have K=1.

If I am right, then the code should be
`pqv_temp = (self.acceptance_param - 1.0) * (
            e - self.energy_state.current_energy) / (
                self.temperature_step)+1.`

Can you please check whether the code is right here? Thanks.

### PR DESCRIPTION
**Pull request by sethtroisi**
#### Reference issue

Fixes: https://github.com/scipy/scipy/issues/10892

#### What does this implement/fix?

**Change dual_annealing accept_reject calculation.**
@deng-cy noticed that we probably have +1 inside a parenthesis where it probably doesn't belong and I'm inclined to agree (because in several tests pqv_temp_old is never positive (e.g. we never accept))

#### Additional information

I tested with the Rastrigin example from [the docs](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.dual_annealing.html)
```
import numpy as np
from scipy.optimize import dual_annealing, minimize

#func = lambda x: np.sum(x*x - 10*np.cos(x)) + 10*np.size(x)
#lower = [-0.5, -0.4, -0.3, -0.2, 0.5]
#upper = [   1,    2,    3,    4,   5]

func = lambda x: np.sum(x*x - 10*np.cos(2*np.pi*(x))) + 10*np.size(x)
lower = [-5.12] * 10
upper = [5.12] * 10

bounds = list(zip(lower, upper))

ret = dual_annealing(func,bounds=bounds,local_search_options={"method":"SLSQP"})
print (ret)
```
Both new and old code find the all zero solution in roughly the same number of iterations so I'm unsure what better test I could be performing.

### CODE REVIEW

File: scipy/optimize/_dual_annealing.py [hunk 1]
```diff
@@ -256,9 +256,13 @@ def __init__(self, acceptance_param, visit_dist, func_wrapper,
 
     def accept_reject(self, j, e, x_visit):
         r = self._rand_gen.uniform()
-        pqv_temp = (self.acceptance_param - 1.0) * (
+        pqv_temp_old = (self.acceptance_param - 1.0) * (
             e - self.energy_state.current_energy) / (
                 self.temperature_step + 1.)
+        assert pqv_temp_old <= 0
```
Comment (sethtroisi) [inline scipy/optimize/_dual_annealing.py:262]: This code will be deleted before I submit it's just here to demonstrate how absurd the old code is.

**Comment by sethtroisi** [comment]
There's an error in scipy/linalg/tests/test_lapack.py:1759: in test_heequb and I've been seeing it locally but I believe it's unrelated (I saw it in other recent PRs too?)

**Comment by rgommers** [comment]
> There's an error in scipy/linalg/tests/test_lapack.py:1759: in test_heequb and I've been seeing it locally but I believe it's unrelated (I saw it in other recent PRs too?)

That's right. @ilayn I believe you were still figuring out if that was gone or now flaky?

**Comment by sethtroisi** [comment]
I read the [Core Developer Guide](http://scipy.github.io/devdocs/dev/core-dev/index.html) and I'm going to start labeling small bugs I work on as part of the next milestone, which seems to be inline with the guide's process

**Comment by rgommers** [comment]
> I read the [Core Developer Guide](http://scipy.github.io/devdocs/dev/core-dev/index.html) and I'm going to start labeling small bugs I work on as part of the next milestone, which seems to be inline with the guide's process

That's fine (and helpful) indeed. I'd suggest only for small things indeed that are likely to get merged quickly. 

What I typically do is set the milestone at merge time.

**Comment by sethtroisi** [comment]
This contain a commit from #12347

**Comment by sethtroisi** [comment]
My understanding of the situation is this isn't really doing dual_annealing but a greedy hill climbing (because we never (rarely?) accept higher energy states. Given this is quite a change (even if it's an improvement this should probably get a release note (I forget this process so let me go explore that process again)

**Comment by rgommers** [comment]
> Given this is quite a change (even if it's an improvement this should probably get a release note (I forget this process so let me go explore that process again)

That's a matter of editing https://github.com/scipy/scipy/wiki/Release-note-entries-for-SciPy-1.6.0 post-merge.

**Review by rgommers — APPROVED** [review]
I went through this again, after the summary in https://github.com/scipy/scipy/pull/12347#issuecomment-659985445. This does look good, and there's agreement in gh-10892 as well that the fix is correct.

So let's get this in. Thanks for the patience @sethtroisi!

**Comment by sethtroisi** [comment]
I added a note in [Release-note-entries-for-SciPy-1.6.0#other-changes](https://github.com/scipy/scipy/wiki/Release-note-entries-for-SciPy-1.6.0#other-changes)

**Pull request merged**